backends/p4tools/common/core/abstract_execution_state.cpp

```diff
@@ -101,7 +101,7 @@ const IR::Expression *AbstractExecutionState::convertToComplexExpression(
         IR::Vector<IR::Expression> components;
         const auto *elementType = resolveType(ts->elementType);
         for (size_t idx = 0; idx < ts->getSize(); idx++) {
-            auto ref = new IR::ArrayIndex(elementType, parent, new IR::Constant(idx));
+            auto ref = new IR::ArrayIndex(elementType, parent, new IR::Constant(uint64_t(idx)));
```

Comment:
Out of curiosity, was the original code resolved to a wrong overload, if so, what platform is it?

Reply:
This is the typical Mac problem. `size_t` is `unsigned long` there, not `unsigned long long` as e.g. on Linux.